rskj-core/src/main/java/co/rsk/core/RskFactory.java

```diff
@@ -217,13 +217,12 @@ public EthereumChannelInitializerFactory getEthereumChannelInitializerFactory(Ch
     }
 
     @Bean
-    public EthHandlerFactoryImpl.RskWireProtocolFactory getRskWireProtocolFactory(ApplicationContext ctx,
-                                                                                  PeerScoringManager peerScoringManager,
+    public EthHandlerFactoryImpl.RskWireProtocolFactory getRskWireProtocolFactory(PeerScoringManager peerScoringManager,
+                                                                                  MessageHandler messageHandler,
                                                                                   Blockchain blockchain,
                                                                                   RskSystemProperties config,
                                                                                   CompositeEthereumListener ethereumListener){
-        // TODO: break MessageHandler circular dependency
-        return () -> new RskWireProtocol(config, peerScoringManager, ctx.getBean(MessageHandler.class), blockchain, ethereumListener);
+        return () -> new RskWireProtocol(config, peerScoringManager, messageHandler, blockchain, ethereumListener);
```

Comment:
Here you still need the lambda function? I believe that now with the removal of ctx.getBean is no longer necessary

Reply:
Yes, we need this one b/c we are creating an instance of `EthHandlerFactoryImpl.RskWireProtocolFactory` here

Reply:
:+1: my bad